TomeOfKnowledge.playground/Pages/Exploring Expressions.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,50 @@
+
+import DiceKit
+
+//: Lets start with the negative-sided die.
+
+let negativeDie = d(-12)
+
+//: A die with negative sides has results in the range `-sides...-1`. We have two ways to represent an expression as a string.
+
+let normalString = String(negativeDie)
+let verboseString = String(reflecting: negativeDie)
+
+//: `normalString` is the string returned from `negativeDie.description`. Our `description`s are intended to be readable and closely match regular dice notation. This also happens to be the text that playgrounds display to the right.
+//: `verboseString` is what is returned from `negativeDie.debugDescription`. These strings tend to do less to make the result readable, and are intended to show exactly what types are used. 
+
+//: When we look at the `description` of a die roll, we seperate the rolled value from the total number of sides with a |. So a d6 that rolled a 3 would be written as 3|6. Or with our special negative die: 
+
+let negativeResult = negativeDie.roll()
+
+//: Multiplication expressions are particularly interesting to look at. We will make one multiplication expression and see what happens when we make a third multiplication expression from the first one and a die.
+let oneD2 = 1 * d(2)
+let d3 = d(3)
+
+//: `oneD2` is of the type `MultiplicationExpression<Constant, Die>` because it is made by multiplying a `Consant` (the expression type `Int` is converted to) by a `Die`.
+
+oneD2 is MultiplicationExpression<Constant, Die>
+
+//: We can make a new multiplication expression with these two expressions. 
+
+let oneD2d3 = oneD2 * d3
+
+//: This expression works like a "normal" multiplication expression, but the multiplier is one half the time, and two the other half of the time. When we `evaluate()` a multiplication expression, we can see what each side of the expression evaluated to. First, an easy case:
+
+let twoD20 = 2 * d(20)
+let twoD20result = twoD20.evaluate()
+
+//: This result type has a `.multiplierResult` and  .`multiplicandResults`. The multiplier is always 2, because it is a constant. The multiplicand is evaluated and summed n number of times to give the result, where n is the multiplier result. We seperate the multiplier and multiplicand with the symbol `*>` in the description, which says that the result of the left hand side is the length of the right hand side.
+
+let oneD2d3result = oneD2d3.evaluate()
+
+//: Now we see the result of `oneD2d3`, which is made up of three terms sperated by `*>`. The leftmost term is the multiplier for 1d2. It must always be 1. The middle term is either a 1|2 or a 2|2, depending on what the d2 rolled. Finally, the rightmost term is either a single d3 roll or the sum of two d3 rolls, depending on what the d2 rolled. The result of the expression is always the sum of the rightmost term.
```

Comment:
Yeah, this one is weird because I just plopped a new line without talking about it first. But elsewhere on this page I did save remarks for after a line. So I went through and made more changes to be consistent. I added as a new commit instead of amending so that the style changes can be compared.